Gordon360/Services/AccountService.cs

```diff
@@ -85,6 +85,25 @@ public AccountViewModel GetAccountByUsername(string username)
             return account;
         }
 
+        /// <summary>
+        /// Fetches the account record with the specified username. Inclusive of non-gordon accounts made by Rec-IM
+        /// </summary>
+        /// <param name="username">The AD username associated with the account.</param>
+        /// <returns>account information</returns>
+        public AccountViewModel GetUnaffiliatedAccountByUsername(string username)
+        {
+            var account = _context.ACCOUNT.FirstOrDefault(x => x.AD_Username == username);
+            if (account == null)
+            {
+                return new AccountViewModel()
+                {
+                    ADUserName = username,
+                    AccountType = "Unaffiliated"
+                };
+            }
+            return account;
+        }
+
```

Comment:
I'm not sure where you use this, but since this business logic is specific to RecIM, I don't think it belongs in `AccountService`.

You can just wrap `AccountService.GetAccountByUsername()` with a try catch block and return this 'Unaffiliated' account in the case where that method throws a `ResourceNotFoundException`. For example, you could put the following in the RecIM ParticipantService:

```suggestion
        /// <summary>
        /// Fetches the account record with the specified username. Inclusive of non-gordon accounts made by Rec-IM
        /// </summary>
        /// <param name="username">The AD username associated with the account.</param>
        /// <returns>account information</returns>
        public AccountViewModel GetUnaffiliatedAccountByUsername(string username)
        {
            AccountViewModel account;
            try {
                account = _accountService.GetAccountByUsername(username);
            } catch (ResourceNotFoundException e) {
                account = new AccountViewModel()
                    {
                        ADUserName = username,
                        AccountType = "Unaffiliated"
                    };
            }
            return account;
        }
```

Reply:
I thought of this later, but forgot to actually make this change. Good catch :)

Reply:
I ended up just moving the service to participant service